### PR TITLE
Bump version to 0.24.2

### PR DESCRIPTION
**Pull request by tgaddair**
This is a cherry-pick from the `hotfix-0242` branch.

### CODE REVIEW

**Review by nvcastet — APPROVED** [review]

**Comment by EnricoMi** [comment]
Are you going to remove 8a12dc97 from this branch, or should the version be minor-incremented as that commit adds features.

**Review by EnricoMi — CHANGES_REQUESTED** [review]
The new features are not in the v0.24.2 change log but they are still part of this branch... Should that commit be removed from this branch or added to the changelog and version incremented to v0.25.0?

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,8 +22,11 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
 ### Fixed
 
-- [Setup] Require fsspec >= 2010.07.0 ([#3451](https://github.com/horovod/horovod/pull/3451))
+## [v0.24.2] - 2022-03-10
+
+### Fixed
 
+- [Setup] Require fsspec >= 2010.07.0 ([#3451](https://github.com/horovod/horovod/pull/3451))
```
Comment (EnricoMi) [inline CHANGELOG.md:29]: I would remove the `[Setup]` as this does not reflect common format of change log entries.

**Review by EnricoMi — CHANGES_REQUESTED** [review]

**Comment by EnricoMi** [comment]
This has already been tagged and is about to be released: https://github.com/horovod/horovod/actions/runs/1964873218

**Comment by maxhgerlach** [comment]
@EnricoMi, the 0.24.2 release was made from a separate branch https://github.com/horovod/horovod/commits/hotfix-0242

**Comment by EnricoMi** [comment]
@maxhgerlach thanks for pointing me to it, haven't seen it though the descriptions says it, then everything is alright.

**Pull request merged**